### ISSUE TITLE
No way to patch the YouTube app?

### ISSUE THREAD

**Issue by SevastianGit** [reisxd/TizenTube#13]
If we're already capable of installing our own apps, would there be no way to extract the YouTube app file/package and then patch/modify it? Similar to the basically equivalent [LGTV homebrew project](https://github.com/webosbrew/youtube-webos).

**Comment by reisxd** [comment]
YouTube for TV app is just a web page. The YouTube LG modification works with root, and there's something called user scripts. User scripts are just extensions that gets injected to the YT web app. Tizen does not have this, nor a known method to root. 


There are other ways to simplify the installation of the app, like creating an app that connects to the SDB TCP socket and launching the YT app in debugging mode. Problem with that is we'd have to maintain two seperate codebases, one for TVs that have NaCl support, one for TVs that have Webassembly support. My TV is from 2017 and I can't test Webassembly on my TV.

Also, writing native code and maintaining it for both platforms is not the greatest idea. 

Lastly, TizenTube has the same features that YT LG has. The code is forked from YT LG. 

TLDR: Not possible due to complexity.

**Comment by SevastianGit** [comment]
About you saying that it requires root, if to believe their [homebrew store page](https://repo.webosbrew.org/apps/youtube.leanback.v4) it does not require root and can be installed without issues just by using the official developer mode without gaining root access.
I am not sure how old my TV is, but if you ever wish to test Webassembly stuff, I could try checking how my TV is and would be glad to help if needed

**Comment by SevastianGit** [comment]
And yeah, I looked into it better, and it is pretty interesting that these apps are mostly just webpages that load from youtube.com/tv

**Comment by notengo13** [comment]
try [samygo](https://forum.samygo.tv) to root your TV if your firmware is supoorted more info on samygo.

**Comment by notengo13** [comment]
> > try [samygo](https://forum.samygo.tv) to root your TV if your firmware is supoorted more info on samygo.
> 
> Huh, very interesting, but if there's not much homebrew, what really can we use root for? (Any way to at least get ad-free YouTube without need to run server?)

Check this thread [J root] (https://forum.samygo.tv/viewtopic.php?p=99454#p99454) where you can find more info what you can do with root and then you need to check if you can root your TV. It depends on what firmware you have.
For legacy's model there was some project also which blocked ads on YouTube. Try to contact with developers from samygo and maybe they can help you to run server on rooted Tizen TV.

**Comment by SevastianGit** [comment]
> YouTube for TV app is just a web page. The YouTube LG modification works with root, and there's something called user scripts. User scripts are just extensions that gets injected to the YT web app. Tizen does not have this, nor a known method to root.
> 
> There are other ways to simplify the installation of the app, like creating an app that connects to the SDB TCP socket and launching the YT app in debugging mode. Problem with that is we'd have to maintain two seperate codebases, one for TVs that have NaCl support, one for TVs that have Webassembly support. My TV is from 2017 and I can't test Webassembly on my TV.
> 
> Also, writing native code and maintaining it for both platforms is not the greatest idea.
> 
> Lastly, TizenTube has the same features that YT LG has. The code is forked from YT LG.
> 
> TLDR: Not possible due to complexity.

As far as I can see, it seems that you have officially made it standalone now? Legend! 🙏🙏❤️‍🔥
